### PR TITLE
[BUG] Enforce forecasting demo dataset whitelist to fix tuple misinterpretation and discovery bugs

### LINKED ISSUE #382: [BUG] load_dataset misinterprets tuple-based non-forecasting demo datasets as forecasting y/exog
## Summary
`Executor.load_dataset()` currently assumes tuple-returning demo loaders use forecasting semantics `(y, X)`.

Because demo datasets are auto-discovered from all `sktime.datasets.load_*` functions, classification datasets like `basic_motions`, `arrow_head`, and `gunpoint` are also exposed. Those loaders return supervised-learning style tuples where features come first and class labels come second.

## Reproduction
On current `main`, these datasets appear in the discovered demo dataset map and can be loaded through `Executor.load_dataset()`.

For example, `basic_motions` currently returns:
- `data`: feature `DataFrame`
- `exog`: class-label `ndarray`

That means labels are surfaced as forecasting exogenous variables.

## Why it matters
This is a core workflow correctness bug. MCP clients can see these datasets as available demos, but the runtime silently interprets them with forecasting semantics, which is misleading for downstream tool calls.

## Suggested fix
Fail fast for tuple-based supervised datasets that are not forecasting-compatible, with a clear error explaining that the dataset is not supported by the current forecasting-oriented high-level workflows.

### LINKED ISSUE #384: [BUG] demo dataset discovery advertises non-forecasting datasets that high-level workflows cannot use
## Summary
The runtime currently auto-discovers all `sktime.datasets.load_*` functions and advertises them through demo dataset discovery.

After rejecting tuple-based supervised datasets in `load_dataset`, discovery still lists non-forecasting datasets like `basic_motions`, `arrow_head`, and `gunpoint` as available demos.

## Reproduction
On current code, `list_available_data_tool(is_demo=True)` includes datasets such as:
- `basic_motions`
- `arrow_head`
- `gunpoint`

But those datasets are not actually usable in the current forecasting-oriented high-level workflow.

## Why it matters
This leaves the MCP surface misleading at the discovery layer. Agents are told these datasets are valid demo choices, only to hit a failure later when they try to load them.

## Suggested fix
Filter discovery so only demo datasets compatible with the current forecasting-oriented workflow are advertised.

### LINKED ISSUE #388: [BUG] export_code accepts unsupported demo datasets for fit examples and generates misleading forecasting code
## Summary
`export_code_tool(..., include_fit_example=True, dataset=...)` currently accepts demo datasets that are not compatible with the forecasting-oriented fit example template.

## Reproduction
On current `main`, this succeeds:

```python
export_code_tool(handle, include_fit_example=True, dataset="basic_motions")
```

and generates code like:

```python
from sktime.datasets import load_basic_motions
y = load_basic_motions()
model.fit(y)
```

## Why it matters
`basic_motions` is not a forecasting demo dataset in the current high-level workflow. Generated code is trusted output, so producing misleading example code is worse than rejecting the request.

## Suggested fix
When `include_fit_example=True`, validate that the requested dataset is compatible with the forecasting-oriented fit example template and return a clear error for unsupported demo datasets.

### PR DESCRIPTION
**Pull request by SHIVANSH-ux-ys**
#### Reference Issues/PRs
Fixes #382
Fixes #384
Fixes #388

#### What does this implement/fix? Explain your changes.

The MCP layer currently assumes all `sktime.datasets` are forecasting datasets, which breaks when LLM agents attempt to use tuple-based supervised datasets like `basic_motions`. This PR hardens the demo dataset API and resolves 3 related bugs:

1. **Discovery (#384):** Filters `list_available_data_tool(is_demo=True)` to only advertise known forecasting datasets.
2. **Loading (#382):** Safely rejects unsupported datasets in `load_dataset` to prevent backwards `y, X` assignment.
3. **Code Generation (#388):** Prevents `export_code_tool` from generating broken `fit` examples for non-forecasting datasets.

It implements this via a `FORECASTING_DEMO_DATASETS` constant whitelist in `executor.py`.

#### Does your contribution introduce a new dependency? If yes, which one?
No.

#### What should a reviewer concentrate their feedback on?
Reviewers can verify the implementation of `FORECASTING_DEMO_DATASETS` in `executor.py` and ensure the fallback validations in `load_dataset` and `export_code_tool` handle exceptions gracefully without crashing the server.

#### Any other comments?
This is another piece of the ESoC 2026 stability push! It prevents agents from hallucinating code on datasets the workflow doesn't fully support yet.

#### PR checklist
- [x] I have read the [CONTRIBUTING](https://github.com/sktime/sktime/blob/main/CONTRIBUTING.md) guide.
- [x] I have checked that my PR does not duplicate an existing PR.
- [x] I have checked that there is an existing issue for this PR, if applicable.
- [x] My code follows the project's coding standards.

### CODE REVIEW

**Comment by SHIVANSH-ux-ys** [comment]
Closing this in favor of the existing PRs for these issues. Apologies for the duplicate effort!
